framework/src/main/java/org/checkerframework/framework/ajava/ExpectedTreesVisitor.java

```diff
@@ -0,0 +1,421 @@
+            }
+
+            if (last.getType().getKind() == Kind.ANNOTATED_TYPE) {
+                AnnotatedTypeTree annotatedType = (AnnotatedTypeTree) last.getType();
+                if (annotatedType.getUnderlyingType().getKind() == Kind.ARRAY_TYPE) {
+                    trees.remove(annotatedType);
+                    trees.remove(annotatedType.getUnderlyingType());
+                }
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public Void visitMethodInvocation(MethodInvocationTree tree, Void p) {
+        Void result = super.visitMethodInvocation(tree, p);
+        // In a method invocation like myObject.myMethod(), the method invocation stores
+        // myObject.myMethod as its own MemberSelectTree which has no corresponding JavaParserNode.
+        if (tree.getMethodSelect().getKind() == Kind.MEMBER_SELECT) {
+            trees.remove(tree.getMethodSelect());
+        }
+
+        return result;
+    }
+
+    @Override
+    public Void visitModifiers(ModifiersTree tree, Void p) {
+        // Don't add ModifierTrees or children because they have no corresponding JavaParser node.
+        return null;
+    }
+
+    @Override
+    public Void visitNewArray(NewArrayTree tree, Void p) {
+        // Skip array initialization because it's not implemented yet.
+        return null;
+    }
+
+    @Override
+    public Void visitNewClass(NewClassTree tree, Void p) {
+        defaultAction(tree);
+
+        if (tree.getEnclosingExpression() != null) {
+            tree.getEnclosingExpression().accept(this, p);
+        }
+
+        tree.getIdentifier().accept(this, p);
+        for (Tree typeArgument : tree.getTypeArguments()) {
+            typeArgument.accept(this, p);
+        }
+
+        for (Tree arg : tree.getTypeArguments()) {
+            arg.accept(this, p);
+        }
+
+        if (tree.getClassBody() == null) {
+            return null;
+        }
+
+        // Anonymous class bodies require special handling. There isn't a corresponding JavaParser
+        // node, and synthetic constructors must be skipped.
+        ClassTree body = tree.getClassBody();
+        visit(body.getModifiers(), p);
+        visit(body.getTypeParameters(), p);
+        visit(body.getImplementsClause(), p);
+        for (Tree member : body.getMembers()) {
+            // Constructors cannot be declared in an anonymous class, so don't add them.
+            if (member.getKind() == Kind.METHOD) {
+                MethodTree methodTree = (MethodTree) member;
+                if (methodTree.getName().contentEquals("<init>")) {
+                    continue;
+                }
+            }
+
+            member.accept(this, p);
+        }
+
+        return null;
+    }
+
+    @Override
+    public Void visitLambdaExpression(LambdaExpressionTree tree, Void p) {
+        for (VariableTree parameter : tree.getParameters()) {
+            // Parameter types might not be specified for lambdas. When not specified, JavaParser
```

Comment:
Make the actor clear in the first sentence:  it's the programmer.
Also, what does javac do in this case?  The second sentence mentions JavaParser.